### DISCUSSION TITLE
Attempting to set parameter  <Use a descriptive title here>

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#28068] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi, 
I have a moose-based app in which I have added a function in a material property as followimg:
in header file:
 const Function * Tn;
in source file:
```
 params.addParam<FunctionName>("mass_function", "The mass flow rate as a ");
```
followed by a simple calcualtion in the code.
in input file:
Tn = 'function name'
it compiles successfully but when i am running, i shows following error and stops the run.
```
*** ERROR ***
Attempting to set parameter "Tn_o" with type (FunctionName)
but the parameter already exists as type (double)
```
Thanks in advance for any help

**Comment by Bearunner** [comment]


**Reply by Rahim-Habibi** [reply]
I have got the same error.

**Comment by GiudGiud** [comment]
Hello

Do you happen to have
```
 params.addParam<Real>("mass_function", "The mass flow rate as a ");
```
in the validParams as well as the FunctionName version?
And if not, what does your class derive from? and does any of the parent have this parameter?

Guillaume

**Reply by Rahim-Habibi** [reply]
Hi @GiudGiud,

No i do not have Real version, only have FunctionName version.
it is derived from "CZMComputeLocalTractionTotalBaseRSF3D" and the parant does not have the parameter.

**Comment by Rahim-Habibi** [comment]
Hi @GiudGiud 
I have edited the header and source like the following: 
in header file: 
```
public:
  const FunctionName functionName() const;    //returns The function name

protected:
  const Function & _Tn_o;                                   //Function to evaluate 

```
in  source file:
```
 params.addParam<FunctionName>("mass_function", "The mass flow rate as a ");
.
.
.
classname::classname(const InputParameters & parameters)
  : CZMComputeLocalTractionTotalBaseRSF3D(parameters),
   _Tn_o(getFunction("Tn_o")),
.
.
.
const FunctionName
classname::functionName() const
{
  return _Tn_o.name();
}
```
But, i got the same error again. 
Thanks for your help

**Reply by GiudGiud** [reply]
this should be
```
   _Tn_o(getFunction("mass_function")),
```
since it uses the name of the parameter

**Reply by Rahim-Habibi** [reply]
Thanks @GiudGiud 
I have followed this ticket #25688 and the issue is resolved. 
in next step, I want to check the magnitude of the function after run in paraview and i call it in input file: output_properties = 'Tn_o',
but "Tn_o" does not show up in paraview. It is not added for automatic output at all. 
Setting Up
  Setting Mesh
    Reading Mesh..                                                                       [ 16.43 s] [  388 MB]
  Finished Setting Mesh                                                                  [ 18.57 s] [  390 MB]
The following total 1 aux variable:
  traction_normal
is added for automatic output by MaterialOutputAction.
  Initializing

**Reply by GiudGiud** [reply]
Hello

There is no Automated function output

Either use a GenericFunctionMaterial to output the function as a material property using the material property output

Or use a FunctionAux to output it as an auxiliary variable

**Reply by Rahim-Habibi** [reply]
Thnaks @GiudGiud 

I already tried in those way and data shows up as an auxvariable.
But waht i mean is that "Tn_o" used like stress in material block, so since i can output stress in exodus file i can output "Tn_o" as well, right?
Actually "Tn_o" is used for initialization of the model, so i want to check if initilazation is done properly? because this app is part of a serial coupling and i need to prove that the results of the app A is transferred to app B properly.

**Reply by GiudGiud** [reply]
> so since i can output stress in exodus file i can output "Tn_o" as well, right?

If Tn_0 is a function then no you can't. 

> so i want to check if initilazation is done properly

You can use an auxvariable and execute the FunctionAux on INITIAL
